warp/cli/commands.py

```diff
@@ -27,29 +29,99 @@ async def send_req(method, url, tx: Optional[Union[str, Dict[str, Any]]] = None)
+    contract_base,
+    program_info: dict,
+    address,
+    function,
+    evm_inputs,
+    network: str,
+    gateway_url: str = "",
 ):
     evm_calldata = get_evm_calldata(program_info["sol_abi"], function, evm_inputs)
     cairo_calldata = get_cairo_calldata(evm_calldata)
-    starknet_invoke(contract_base, address, cairo_calldata, network)
+    starknet_invoke(contract_base, address, cairo_calldata, network, gateway_url)
     return True
 
 
+async def _call(
+    contract_base,
+    program_info: dict,
+    address,
+    function,
+    evm_inputs,
+    network: str,
+    gateway_url: str = "",
+):
+    evm_calldata = get_evm_calldata(program_info["sol_abi"], function, evm_inputs)
+    cairo_calldata = get_cairo_calldata(evm_calldata)
+    starknet_call(contract_base, address, cairo_calldata, network, gateway_url)
+    return True
+
+
+def starknet_call(
+    contract_base,
+    address,
+    cairo_calldata: Sequence[int],
+    network: str,
+    gateway_url: str = "",
+):
+    abi = f"{contract_base}_abi.json"
+    inputs = " ".join(map(str, cairo_calldata))
+    if gateway_url:
+        print(
+            os.popen(
+                f"starknet call "
+                f"--address {address} "
+                f"--feeder_gateway_url={gateway_url} "
+                f"--abi {abi} "
+                f"--function __main "
+                f"--inputs {inputs} "
+            ).read()
+        )
+    else:
+        print(
+            os.popen(
+                f"starknet call "
+                f"--address {address} "
+                f"--abi {abi} "
+                f"--function __main "
+                f"--inputs {inputs} "
+                f"--network {network} "
+            ).read()
+        )
+
+
 def starknet_invoke(
-    contract_base, address, cairo_calldata: Sequence[int], network: str
+    contract_base,
+    address,
+    cairo_calldata: Sequence[int],
+    network: str,
+    gateway_url: str = "",
 ):
     abi = f"{contract_base}_abi.json"
     inputs = " ".join(map(str, cairo_calldata))
-    print(
-        os.popen(
-            f"starknet invoke "
-            f"--address {address} "
-            f"--abi {abi} "
-            f"--function __main "
-            f"--inputs {inputs} "
-            f"--network {network} "
-        ).read()
-    )
+    if gateway_url:
+        print(
+            os.popen(
+                f"starknet invoke "
+                f"--address {address} "
+                f"--feeder_gateway_url={gateway_url} "
+                f"--abi {abi} "
+                f"--function __main "
+                f"--inputs {inputs} "
+            ).read()
+        )
+    else:
+        print(
+            os.popen(
+                f"starknet invoke "
+                f"--address {address} "
+                f"--abi {abi} "
+                f"--function __main "
+                f"--inputs {inputs} "
+                f"--network {network} "
+            ).read()
+        )
```

Comment:
Minor, but maybe we can just extract `network_param = "--network {network}" \/ "--feeder={gateway}"` instead of duplicating the whole call?

Also, when `os.popen` fails, it doesn't report errors, so it would be nice to use something like [subprocess.check_output](https://docs.python.org/3/library/subprocess.html?highlight=subprocess%20run#subprocess.check_output)